staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -887,11 +907,12 @@ func (r *Request) request(ctx context.Context, fn func(*http.Request, *http.Resp
 				fn(req, resp)
 			}
 
-			if r.retry.IsNextRetry(ctx, r, req, resp, err, isErrRetryableFunc) {
+			if r.retry.IsNextRetry(ctx, r, req, resp, closer, err, isErrRetryableFunc) {
 				return false
 			}
 
-			f(req, resp)
+			respCopy.Body = closer.respCopy
```

Comment:
if we replace resp.Body internally in DrainAndClose(), I don't think we need to do this